x/exchange/types/mem_order.go

```diff
@@ -29,193 +29,116 @@ func (typ MemOrderType) String() string {
+	ordererAddr sdk.AccAddress, isBuy bool, price sdk.Dec, qty, openQty, remainingDeposit sdk.Int, source OrderSource) *MemOrder {
 	return &MemOrder{
-		typ:              OrderSourceMemOrder,
-		ordererAddr:      ordererAddr,
-		isBuy:            isBuy,
-		price:            price,
-		qty:              qty,
-		openQty:          openQty,
-		remainingDeposit: DepositAmount(isBuy, price, openQty),
-		executedQty:      utils.ZeroDec,
-		paid:             utils.ZeroDec,
-		received:         utils.ZeroDec,
-		fee:              utils.ZeroDec,
-		source:           source,
+		Type:             OrderSourceMemOrder,
+		OrdererAddress:   ordererAddr,
+		Source:           source,
+		IsBuy:            isBuy,
+		Price:            price,
+		Quantity:         qty,
+		OpenQuantity:     openQty,
+		RemainingDeposit: remainingDeposit,
+		MatchState:       NewMatchState(),
 	}
 }
 
 func (order *MemOrder) String() string {
 	isBuyStr := "buy"
-	if !order.isBuy {
+	if !order.IsBuy {
 		isBuyStr = "sell"
 	}
 	return fmt.Sprintf(
-		"{%s %s %s %s}",
-		order.typ, isBuyStr, order.price, order.openQty)
-}
-
-func (order *MemOrder) Type() MemOrderType {
-	return order.typ
-}
-
-func (order *MemOrder) Order() Order {
-	return *order.order
-}
-
-func (order *MemOrder) Source() OrderSource {
-	return order.source
-}
-
-func (order *MemOrder) OrdererAddress() sdk.AccAddress {
-	return order.ordererAddr
-}
-
-func (order *MemOrder) IsBuy() bool {
-	return order.isBuy
-}
-
-func (order *MemOrder) Price() sdk.Dec {
-	return order.price
-}
-
-func (order *MemOrder) Quantity() sdk.Dec {
-	return order.qty
-}
-
-func (order *MemOrder) OpenQuantity() sdk.Dec {
-	return order.openQty
-}
-
-func (order *MemOrder) RemainingDeposit() sdk.Dec {
-	return order.remainingDeposit
-}
-
-func (order *MemOrder) ExecutedQuantity() sdk.Dec {
-	return order.executedQty
-}
-
-func (order *MemOrder) Paid() sdk.Dec {
-	return order.paid
-}
-
-func (order *MemOrder) PaidWithoutFee() sdk.Dec {
-	if order.fee.IsNegative() {
-		return order.paid.Sub(order.fee)
-	}
-	return order.paid
-}
-
-func (order *MemOrder) Received() sdk.Dec {
-	return order.received
+		"{%s %s %s %s/%s}",
+		order.Type, isBuyStr, order.Price, order.MatchState.executedQty, order.OpenQuantity)
 }
 
-func (order *MemOrder) Fee() sdk.Dec {
-	return order.fee
-}
-
-func (order *MemOrder) IsMatched() bool {
-	return order.isMatched
-}
-
-func (order *MemOrder) ExecutableQuantity() sdk.Dec {
-	executableQty := order.openQty.Sub(order.executedQty)
-	if order.isBuy {
-		return sdk.MinDec(executableQty, order.remainingDeposit.QuoTruncate(order.price))
+func (order *MemOrder) ExecutableQuantity() sdk.Int {
+	executableQty := order.OpenQuantity.Sub(order.executedQty)
+	if order.IsBuy {
+		return utils.MinInt(
+			executableQty,
+			order.RemainingDeposit.ToDec().QuoTruncate(order.Price).TruncateInt())
 	}
-	return sdk.MinDec(executableQty, order.remainingDeposit)
+	return utils.MinInt(executableQty, order.RemainingDeposit.ToDec().TruncateInt())
```

Comment:
Why do we need to truncate after ToDec()? (`RemainingDeposit` is already Int type.)
Can we use the following?
```
return utils.MinInt(executableQty, order.RemainingDeposit)
```

Reply:
That's right. It was my mistake